### DISCUSSION TITLE
How do I get Application Commands (Slash commands) to work on Visual Studio code?

### DISCUSSION THREAD

**Discussion by Unbeatenchampion** [Rapptz/discord.py#9316] [Q&A] [answered]
I have installed discord.py in visual studio code using the command `py -3 -m pip install -U discord.py` and everything works fine apart from application commands, as it will say it can not find "app_commands" when I try to import it from discord.py 
How do I fix this so that it will import app_commands? did I install wrong discord.py?

(the scribbled out bit is my name, so just ignore that bit 🙂)

![Screenshot 2023-03-29 124413](https://user-images.githubusercontent.com/120485083/228525327-f6d803b1-d438-4ad0-8003-bb22df79b3f1.jpg)
![Screenshot 2023-03-29 124144](https://user-images.githubusercontent.com/120485083/228525329-6cd0771a-713d-4f28-8079-a592c738e691.jpg)

**Comment by hawk-tomy** [comment, marked as answer]
You should use python 3.8 or higher version instead of python 3.7.

- discord.app_commands was added in discord.py v2.0
- discord.py v2.0 require Python 3.8 or higher

**Reply by Unbeatenchampion** [reply]
oh how do I install Python 3.8?

**Reply by Unbeatenchampion** [reply]
Do I just install it from the Python website? Or is there something on VScode i have to do

**Reply by tailoric** [reply]
you will need to install it from the website or depending on your OS from the package manager would be the best. 
Also you should probably use the newest version available for all the features 
https://www.python.org/downloads/

**Reply by Unbeatenchampion** [reply]
Ok thanks 🙂 and I'm using Windows 10

**Reply by Unbeatenchampion** [reply]
So once I have the Newer python it will just work?

**Reply by tailoric** [reply]
you would of course need to install discord.py on the new version of python (or better use a [venv](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment))
But yeah you probably only installed the latest version of discord.py that was compatible with python 3.7 but at least that should make it work (as long as there are no other problems)

**Reply by Unbeatenchampion** [reply]
Ok I will delete 3.7 and try again

**Reply by Unbeatenchampion** [reply]
Yay, it is working thank you everyone! 😄 I am now using python 3.11

**Reply by Einstein197** [reply]
is that for slash commands
